nitro-protocol/contracts/libraries/signature-logic/StrictTurnTaking.sol

```diff
@@ -23,16 +23,11 @@ library StrictTurnTaking {
 
         uint48 turnNum = proof[0].variablePart.turnNum;
 
-        // validating the proof
         for (uint256 i = 0; i < proof.length; i++) {
-            isSignedByMover(fixedPart, proof[i]);
-            requireHasTurnNum(proof[i].variablePart, turnNum);
+            isSignedByMover(fixedPart, (i < proof.length ? proof[i] : candidate));
+            requireHasTurnNum((i < proof.length ? proof[i].variablePart : candidate.variablePart), turnNum);
             turnNum++;
         }
-
-        // validating the candidate
-        isSignedByMover(fixedPart, candidate);
-        requireHasTurnNum(candidate.variablePart, turnNum);
     }
```

Comment:
@geoknee , do you think this change make the code clearer?
For sure, it follows the practice of treating `proof` and `candidate` equally, but does it improves readability?

Reply:
1. I don't think it makes a huge difference. Probably on balance it is slightly **less** readable IMO.
2. I'm not sure it is actually correct? The condition on the loop is for `i<proof.length`, meaning that `candidate` will never be selected by the ternary operator? I think you need to have the loop condition `i<=proof.length` or `i<proof.length+1`.

Reply:
Wow!
How did the tests not see 2)?

Reply:
https://github.com/statechannels/go-nitro/issues/768

Reply:
Treating `proof` and `candidate` in the same way (using ternary operator to specify which one to use) is doubtfully increasing the readability, so I propose we give up on it and left the code as it was.